### PR TITLE
Moves frame rotation to AsyncTask

### PR DESCRIPTION
**Pull request by SimonErm**
Moves frame rotation to BarCodeScannerAsyncTask like it was in RCTCamera to increase Performance.
#1422

### CODE REVIEW

File: android/src/main/java/org/reactnative/camera/tasks/BarCodeScannerAsyncTask.java [hunk 1]
```diff
@@ -41,14 +41,27 @@ protected Result doInBackground(Void... ignored) {
...
+        //no barcode Found
+      }
     } catch (Throwable t) {
       t.printStackTrace();
     }
 
     return result;
   }
-
+  private byte[] rotateImage(byte[]imageData,int width, int height) {
```
Comment (jgfidelis) [inline android/src/main/java/org/reactnative/camera/tasks/BarCodeScannerAsyncTask.java]: this is not a 1:1 copy from the old method.

Why the changes?

File: android/src/main/java/org/reactnative/camera/tasks/BarCodeScannerAsyncTask.java [hunk 2]
```diff
@@ -41,14 +41,27 @@ protected Result doInBackground(Void... ignored) {
...
+    byte[] rotated = new byte[imageData.length];
+    for (int y = 0; y < height; y++) {
+      for (int x = 0; x < width; x++) {
+        rotated[x * height + height - y - 1] = imageData[x + y * width];
```
Comment (jgfidelis) [inline android/src/main/java/org/reactnative/camera/tasks/BarCodeScannerAsyncTask.java]: this may lead to crashes, that is why there was an `if` inside the `for` on the old method.

File: android/src/main/java/org/reactnative/camera/tasks/BarCodeScannerAsyncTask.java (hunk 1 above)
Comment (jgfidelis) [reply android/src/main/java/org/reactnative/camera/tasks/BarCodeScannerAsyncTask.java]: Before it was:
`rotated[x * width + width - y - 1] = imageData[x + y * height];`

File: android/src/main/java/org/reactnative/camera/tasks/BarCodeScannerAsyncTask.java (hunk 2 above)
Comment (jgfidelis) [reply android/src/main/java/org/reactnative/camera/tasks/BarCodeScannerAsyncTask.java]: Please separate into `sourceIx` and `destIx` and check if those are valid, like before

File: android/src/main/java/org/reactnative/camera/tasks/BarCodeScannerAsyncTask.java (hunk 1 above)
Comment (jgfidelis) [reply android/src/main/java/org/reactnative/camera/tasks/BarCodeScannerAsyncTask.java]: Ok, looking again, the difference is because before y and x were `height` and `width` and now they are `width` and `height`. If it produces the same result, ok hehe

Comment (SimonErm) [reply android/src/main/java/org/reactnative/camera/tasks/BarCodeScannerAsyncTask.java]: Yes on last PR i swaped the naming, but now i just copied from RCTCamera and yes it produce the same :-D

File: android/src/main/java/org/reactnative/camera/tasks/BarCodeScannerAsyncTask.java (hunk 2 above)
Comment (SimonErm) [reply android/src/main/java/org/reactnative/camera/tasks/BarCodeScannerAsyncTask.java]: Ok
